data_model/src/lib.rs

```diff
@@ -696,10 +744,21 @@ where
 }
 
 /// This trait marks entity that implement it as identifiable with an `Id` type to find them by.
-pub trait Identifiable: Debug + Clone {
+pub trait Identifiable: Debug {
     /// Type of entity's identification.
     type Id: Into<IdBox> + Debug + Clone + Eq + Ord;
-    /// Type used to register `Identifiable` entity
+
+    fn id(&self) -> &Self::Id;
```

Comment:
just an afterthought, if it weren't for this method we could make a blanket impl like:
```rs
impl<T: RegisteredWith> Identifiable for T::RegisteredWith {
    type Id = <T as Identifiable>::Id;
}
```

Reply:
That's an interesting observation.